For reference, if I try to do the OpenMP-only build but using nvcc_wrapper as cxx compiler (instead of setting as g++), I see compilation errors. That doesn't happen in the nightly job testing the Serial backend but using nvcc_wrapper as cxx compiler. I don't know if we need to support this or not, I can open an issue if so. 

Config:
```
cmake -D CMAKE_BUILD_TYPE=RelWithDebInfo -D CMAKE_CXX_COMPILER=/root/kokkos/bin/nvcc_wrapper -D CMAKE_CXX_STANDARD=20 -D Kokkos_ENABLE_DEBUG=ON -D Kokkos_ENABLE_DEBUG_BOUNDS_CHECK=ON -D Kokkos_ARCH_NATIVE=ON -D Kokkos_ARCH_HOPPER90=ON -D Kokkos_ENABLE_CUDA=OFF -D Kokkos_ENABLE_OPENMP=ON -D Kokkos_ENABLE_COMPILER_WARNINGS=ON -D Kokkos_ENABLE_LARGE_MEM_TESTS=ON -D Kokkos_ENABLE_LIBDL=OFF -D Kokkos_ENABLE_TESTS=ON -G"Unix Makefiles" -S/root/kokkos -B/root/build
```

Errors:
```
nvcc_wrapper has been given GNU extension standard flag -std=gnu++20 - reverting flag to -std=c++20
Error #20015-D: calling a constexpr __host__ function from a __host__ __device__ function is not allowed. The experimental flag '--expt-relaxed-constexpr' can be used to allow this.

Remark: The warnings can be suppressed with "-diag-suppress <warning-number>"

1 error detected in the compilation of "/root/build/core/unit_test/openmp/TestOpenMP_MDSpanConversion.cpp".
make[2]: *** [core/unit_test/CMakeFiles/Kokkos_CoreUnitTest_OpenMP.dir/build.make:832: core/unit_test/CMakeFiles/Kokkos_CoreUnitTest_OpenMP.dir/openmp/TestOpenMP_MDSpanConversion.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[ 54%] Linking CXX executable Kokkos_IncrementalTest_OPENMP
[ 54%] Built target Kokkos_IncrementalTest_OPENMP
/root/kokkos/core/unit_test/TestSpaceAwareAccessor.hpp(93): error: static assertion failed
          static_assert(std::is_empty_v<acc_t>);
          ^
          detected during instantiation of "void test_space_aware_accessor<T,ExecutionSpace,MemorySpace>() [with T=int, ExecutionSpace=Kokkos::OpenMP, MemorySpace=Kokkos::OpenMP::memory_space]" at line 142

/root/kokkos/core/unit_test/TestSpaceAwareAccessor.hpp(93): error: static assertion failed
          static_assert(std::is_empty_v<acc_t>);
          ^
          detected during instantiation of "void test_space_aware_accessor<T,ExecutionSpace,MemorySpace>() [with T=double, ExecutionSpace=Kokkos::OpenMP, MemorySpace=Kokkos::OpenMP::memory_space]" at line 143

/root/kokkos/core/unit_test/TestSpaceAwareAccessor.hpp(93): error: static assertion failed
          static_assert(std::is_empty_v<acc_t>);
          ^
          detected during instantiation of "void test_space_aware_accessor<T,ExecutionSpace,MemorySpace>() [with T=const int, ExecutionSpace=Kokkos::OpenMP, MemorySpace=Kokkos::OpenMP::memory_space]" at line 144

/root/kokkos/core/unit_test/TestSpaceAwareAccessor.hpp(93): error: static assertion failed
          static_assert(std::is_empty_v<acc_t>);
          ^
          detected during instantiation of "void test_space_aware_accessor<T,ExecutionSpace,MemorySpace>() [with T=const double, ExecutionSpace=Kokkos::OpenMP, MemorySpace=Kokkos::OpenMP::memory_space]" at line 145

/root/kokkos/core/unit_test/TestSpaceAwareAccessor.hpp(93): error: static assertion failed
          static_assert(std::is_empty_v<acc_t>);
          ^
          detected during instantiation of "void test_space_aware_accessor<T,ExecutionSpace,MemorySpace>() [with T=double, ExecutionSpace=Kokkos::OpenMP, MemorySpace=Kokkos::AnonymousSpace]" at line 146

/root/kokkos/core/unit_test/TestSpaceAwareAccessor.hpp(93): error: static assertion failed
          static_assert(std::is_empty_v<acc_t>);
          ^
          detected during instantiation of "void test_space_aware_accessor<T,ExecutionSpace,MemorySpace>() [with T=const int, ExecutionSpace=Kokkos::OpenMP, MemorySpace=Kokkos::AnonymousSpace]" at line 147

6 errors detected in the compilation of "/root/build/core/unit_test/openmp/TestOpenMP_SpaceAwareAccessor.cpp".
...
```

Edit: setting `-D CMAKE_CXX_FLAGS='-Werror=all-warnings -Werror' ` in that build also triggers -Werror's, e.g.
```
nvcc_wrapper has been given GNU extension standard flag -std=gnu++20 - reverting flag to -std=c++20
Error #20015-D: calling a constexpr __host__ function from a __host__ __device__ function is not allowed. The experimental flag '--expt-relaxed-constexpr' can be used to allow this.

Remark: The warnings can be suppressed with "-diag-suppress <warning-number>"

1 error detected in the compilation of "/root/build/core/unit_test/openmp/TestOpenMP_View_BasicView.cpp".
make[2]: *** [core/unit_test/CMakeFiles/Kokkos_CoreUnitTest_OpenMP_ViewSupport.dir/build.make:90: core/unit_test/CMakeFiles/Kokkos_CoreUnitTest_OpenMP_ViewSupport.dir/openmp/TestOpenMP_View_BasicView.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
```

--- a/.gitlab/hpsf-gitlab-ci.yml
+++ b/.gitlab/hpsf-gitlab-ci.yml
@@ -1,6 +1,7 @@
 .common-setup:
   stage: test
   variables:
+    OMP_NUM_THREADS: "8"
     OMP_PLACES: "threads"
     OMP_PROC_BIND: "spread"
   before_script:
@@ -54,7 +55,6 @@ NVIDIA-GH200:
   image: nvcr.io/nvidia/cuda:13.1.0-devel-ubuntu24.04
   script:
     - apt-get update && apt-get install -y cmake git python3
-    - export KOKKOS_NUM_THREADS=8
     - export CMAKE_BUILD_PARALLEL_LEVEL=48
     - export ENV_CMAKE_OPTIONS=""
     - export ENV_CMAKE_OPTIONS="${ENV_CMAKE_OPTIONS};-D CMAKE_BUILD_TYPE=${BUILD_TYPE}"
@@ -242,9 +242,6 @@ NVIDIA-GRACE-GRACE:
   image: ubuntu:latest
   script:
     - apt-get update && apt-get install -y cmake git g++
-    - export OMP_NUM_THREADS=8
-    - export OMP_PROC_BIND=close
-    - export OMP_PLACES=cores
     - export CMAKE_BUILD_PARALLEL_LEVEL=48
     - export ENV_CMAKE_OPTIONS=""
     - export ENV_CMAKE_OPTIONS="${ENV_CMAKE_OPTIONS};-D CMAKE_BUILD_TYPE=${BUILD_TYPE}"

--- a/.gitlab/hpsf-gitlab-ci.yml
+++ b/.gitlab/hpsf-gitlab-ci.yml
@@ -2,8 +2,8 @@
   stage: test
   variables:
     OMP_NUM_THREADS: "8"
-    OMP_PLACES: "threads"
-    OMP_PROC_BIND: "spread"
+    OMP_PLACES: "cores"
+    OMP_PROC_BIND: "close"
   before_script:
     - if [ ${CI_PIPELINE_SOURCE} == "schedule" ]; then
         export CDASH_MODEL="Nightly";

